test/python/transpiler/test_sabre_swap.py

```diff
@@ -377,5 +383,839 @@ def test_rejects_too_few_qubits(self):
+        check_map_pass = CheckMap(coupling)
+        check_map_pass.run(cdag)
+        self.assertTrue(check_map_pass.property_set["is_swap_mapped"])
+
+        expected = QuantumCircuit(qreg, creg)
+        expected.h(0)
+        expected.x(1)
+        expected.measure(0, 0)
+        etrue_body = QuantumCircuit(qreg, creg[[0]])
+        etrue_body.x(0)
+
+        efor_body = QuantumCircuit(qreg)
+        efor_body.delay(10, 0)
+        efor_body.barrier(qreg)
+        efor_body.swap(1, 2)
+        efor_body.cx(0, 1)
+        efor_body.swap(1, 2)
+        etrue_body.for_loop(range(3), loop_parameter, efor_body, qreg, [])
+
+        efalse_body = QuantumCircuit(qreg, creg[[0]])
+        efalse_body.y(0)
+        expected.if_else((creg[0], 0), etrue_body, efalse_body, qreg, creg[[0]])
+        expected.measure(qreg, creg)
+
+        self.assertEqual(dag_to_circuit(cdag), expected)
+
+    def test_nested_outer_cnot(self):
+        """test swap with nested if else controlflow construct; swap in outer"""
+        num_qubits = 5
+        qreg = QuantumRegister(num_qubits, "q")
+        creg = ClassicalRegister(num_qubits)
+        coupling = CouplingMap.from_line(num_qubits)
+        qc = QuantumCircuit(qreg, creg)
+        qc.h(0)
+        qc.x(1)
+        qc.measure(0, 0)
+        true_body = QuantumCircuit(qreg, creg[[0]])
+        true_body.cx(0, 2)
+        true_body.x(0)
+
+        for_body = QuantumCircuit(qreg)
+        for_body.delay(10, 0)
+        for_body.barrier(qreg)
+        for_body.cx(1, 3)
+        loop_parameter = None
+        true_body.for_loop(range(3), loop_parameter, for_body, qreg, [])
+
+        false_body = QuantumCircuit(qreg, creg[[0]])
+        false_body.y(0)
+        qc.if_else((creg[0], 0), true_body, false_body, qreg, creg[[0]])
+        qc.measure(qreg, creg)
+
+        dag = circuit_to_dag(qc)
+        cdag = SabreSwap(coupling, "lookahead", seed=82, trials=1).run(dag)
+        check_map_pass = CheckMap(coupling)
+        check_map_pass.run(cdag)
+        self.assertTrue(check_map_pass.property_set["is_swap_mapped"])
+
+        expected = QuantumCircuit(qreg, creg)
+        expected.h(0)
+        expected.x(1)
+        expected.measure(0, 0)
+        etrue_body = QuantumCircuit(qreg, creg[[0]])
+        etrue_body.swap(1, 2)
+        etrue_body.cx(0, 1)
+        etrue_body.x(0)
+
+        efor_body = QuantumCircuit(qreg)
+        efor_body.delay(10, 0)
+        efor_body.barrier(qreg)
+        efor_body.cx(2, 3)
+
+        etrue_body.for_loop(range(3), loop_parameter, efor_body, qreg[[0, 1, 2, 3, 4]], [])
+        etrue_body.swap(1, 2)
+
+        efalse_body = QuantumCircuit(qreg, creg[[0]])
+        efalse_body.y(0)
+        expected.if_else((creg[0], 0), etrue_body, efalse_body, qreg, creg[[0]])
+        expected.measure(qreg, creg[[0, 1, 2, 3, 4]])
+        self.assertEqual(dag_to_circuit(cdag), expected)
+
+    def test_disjoint_looping(self):
```

Comment:
This reminds me there are a couple of tests that I think would be good to have validation on combining control flow with disjoint connectivity. I think two cases in particular of key importance, testing that a disjoint coupling map that has separate connected components in the circuit with control flow in each component works as expected. I think testing this with and without classical bits in the circuit. The difference between clbits shouldn't make a difference in the code anymore since we check if `len(components) > 1` and always throw away the swap maps and rerun after and apply the full layout. But I think for future proofing it's better to test both cases.

We will get this coverage in the last control flow o2 and o3 PR but it's good to test the pass directly too.